### PR TITLE
gitbook: docs update for 1.0.0

### PR DESCRIPTION
**Pull request by droot**

### CODE REVIEW

**Comment by droot** [comment]
@pwittrock @mengqiy @Liujingfang1 I have updated http://docs.kubebuilder.io PTAL (This is a blocker for today's release)

**Comment by pwittrock** [comment]
DO we need to update the simple_resource file to remove the "//+resource" annotation?  Or are we scaffolding that still?  Maybe we should scaffold it just so docs continue to work for now.

File: docs/book/beyond_basics/deploying_controller.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+```bash
+# build the docker image 
+make docker-push IMG=<image-name>
+```
+
+#### Customizing the controller manager manifests using Kustomize
+
+Kubebuilder scaffolds a basic `kustomization.yaml` under `config/default` directory. Current customization includes:
+ - specifies all controller manager resources to be created under specified `namespace`
```
Comment (pwittrock) [inline docs/book/beyond_basics/deploying_controller.md]: nit: match case with 'S'

File: docs/book/quick_start.md [hunk 2]
```diff
@@ -146,4 +154,4 @@ nano -w docs/reference/examples/sloop/sloop.yaml
 ```bash
 kubebuilder docs
```
[the remark this answers is not in the document]
Comment (Liujingfang1) [reply docs/book/quick_start.md]: No, but it gives a `coming soon` message

File: docs/book/basics/project_creation_and_structure.md [hunk 3]
```diff
@@ -47,6 +47,7 @@ A Dockerfile is scaffolded to build a container image for your Manager.
 Kubebuilder creates yaml config for installing the CRDs and related objects under config/.
 
 - config/crds
+- config/rbac
 - config/manager
```
[the remark this answers is not in the document]
Comment (Liujingfang1) [reply docs/book/basics/project_creation_and_structure.md]: `config/samples` is not for installing CDRs. I feel it should not be included in the default kustomization.

File: docs/book/beyond_basics/deploying_controller.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+# Deploy the controller-manager in a Kubernetes cluster
+
+Deploying the controller to a Kubernetes clusters involves following steps:
```
[the remark this answers is not in the document]
Comment (droot) [reply docs/book/beyond_basics/deploying_controller.md]: Done

File: docs/book/quick_start.md [hunk 5]
```diff
@@ -12,11 +12,20 @@ This Quick Start guide will cover:
...
 
 - Install [dep](https://github.com/golang/dep)
+- Install [kustomize](https://github.com/kubernetes-sigs/kustomize)
 - Install [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder)
 
 {% sample lang="bash" %}
 ```bash
-go get -u github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder
+# download the release depending on your OS e.g. linux or darwin
+curl -L -O https://github.com/kubernetes-sigs/kubebuilder/releases/download/v1.0.0/kubebuilder_1.0.0_darwin_amd64.tar.gz
```
[the remark this answers is not in the document]
Comment (droot) [reply docs/book/quick_start.md]: Done

File: docs/book/basics/project_creation_and_structure.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (droot) [reply docs/book/basics/project_creation_and_structure.md]: Its not included in the kustomization.

**Comment by droot** [comment]
Addressed the comments. PTAL.

File: docs/book/beyond_basics/deploying_controller.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+{% endpanel %}
+
+#### Building the docker image
+
+`Makefile` has a target `docker-build` to build the docker image for the
+controller manager. The target supports `IMG` variable.
+
+```bash
+# build the docker image 
+make docker-build IMG=<image-name>
```
Comment (mengqiy) [inline docs/book/beyond_basics/deploying_controller.md]: IMO we should also say that the target will pick it from the environment variable if the variable is not provided.
```
export IMG=<image-name>
make docker-build
```

File: docs/book/beyond_basics/deploying_controller.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+`docker-build` target builds the docker image and also updates the patch file with the provided image name.
+
+#### Pushing the docker image to container registry
+
+`Makefile` has a target `docker-push` to push the docker image to the configured
+container registry. The target supports `IMG` variable.
+
+```bash
+# build the docker image 
+make docker-push IMG=<image-name>
```
Comment (mengqiy) [inline docs/book/beyond_basics/deploying_controller.md]: Same comment as above here.

File: docs/book/getting_started/installation_and_setup.md [hunk 8]
```diff
@@ -1,16 +1,81 @@
...
+## Installing a stable release
+
 Install kubebuilder by downloading the latest stable release from the
-[github repo](https://github.com/kubernetes-sigs/kubebuilder/releases)
-or by running `go get` to get the latest code.
+[github repo](https://github.com/kubernetes-sigs/kubebuilder/releases).
+
+{% sample lang="mac" %}
+```bash
+version=1.0.0 # latest stable version
```
Comment (mengqiy) [inline docs/book/getting_started/installation_and_setup.md]: Not a blocker.
I suggest having a place to hold the latest stable version like https://storage.googleapis.com/kubernetes-release/release/stable.txt
So we don't hardcode the version in the doc. Instead, use curl to get the latest version.

File: docs/book/getting_started/installation_and_setup.md [hunk 9]
```diff
@@ -1,16 +1,81 @@
...
+
+# update your PATH to include /usr/local/kubebuilder/bin
+export PATH=$PATH:/usr/local/kubebuilder/bin
+```
 
-{% sample lang="bash" %}
+{% sample lang="linux" %}
 ```bash
-go get -u github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder
+version=1.0.0 # latest stable version
```
Comment (mengqiy) [inline docs/book/getting_started/installation_and_setup.md]: same here

File: docs/book/quick_start.md [hunk 10]
```diff
@@ -12,11 +12,39 @@ This Quick Start guide will cover:
 {% method %}
 
 - Install [dep](https://github.com/golang/dep)
+- Install [kustomize](https://github.com/kubernetes-sigs/kustomize)
 - Install [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder)
 
-{% sample lang="bash" %}
+{% sample lang="mac" %}
```
Comment (mengqiy) [inline docs/book/quick_start.md]: sample lang="mac" ???

File: docs/book/quick_start.md [hunk 11]
```diff
@@ -12,11 +12,39 @@ This Quick Start guide will cover:
...
 ```bash
-go get -u github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder
+version=1.0.0 # latest stable version
```
Comment (mengqiy) [inline docs/book/quick_start.md]: Same here

File: docs/book/quick_start.md [hunk 12]
```diff
@@ -12,11 +12,39 @@ This Quick Start guide will cover:
...
+tar -zxvf kubebuilder_$version_darwin_$arch.tar.gz
+sudo mv kubebuilder_$version_darwin_$arch /usr/local/kubebuilder
+
+# update your PATH to include /usr/local/kubebuilder/bin
+export PATH=$PATH:/usr/local/kubebuilder/bin
+```
+
+{% sample lang="linux" %}
+```bash
+version=1.0.0 # latest stable version
```
Comment (mengqiy) [inline docs/book/quick_start.md]: And here

File: docs/book/getting_started/installation_and_setup.md (hunk 8 above)
Comment (droot) [reply docs/book/getting_started/installation_and_setup.md]: installation instructions in current form are not the best experience. We will have to replace these with a `install.sh` script which users `curl <script-path> | sh`.

**Comment by mengqiy** [comment]
Doc changes only. Travis flakes a bit on MacOS. Manually merging it so it won't block our release :)

**Pull request merged**